### PR TITLE
Adding Test Runtime folder with Xunit assemblies + runner

### PR DESCRIPTION
**Pull request by joperezr**
cc: @weshaggard @mellinoe @ericstj 

Adding Test runtime folder which we'll need in order to compile test libraries.
For now, this is for netcoreapp only and it is located at /bin/tests/test-runtime/netcoreapp/

I did remove the test-runtime project.json since that won't be needed any longer as tests will compile against the Framework TargetingPack plus the test runtime folder.

### CODE REVIEW

**Comment by mellinoe** [comment]
Looks good to me. I should be able to rebase my work onto this easily.

File: src/Common/test-runtime/XUnit.Runtime.depproj [hunk 1]
```diff
@@ -0,0 +1,25 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Project ToolsVersion="14.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003" DefaultTargets="Build">
+  <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dir.props))\dir.props" />
+  <PropertyGroup>
+    <NuGetTargetMoniker>.NETStandard,Version=v1.7</NuGetTargetMoniker>
+    <IsRuntimeAssembly>false</IsRuntimeAssembly>
```
Comment (weshaggard) [inline src/Common/test-runtime/XUnit.Runtime.depproj]: IsRuntimeAssembly hack should also get fixed when we start tweaking the binplacing targets.

File: src/Common/test-runtime/XUnit.Runtime.depproj [hunk 2]
```diff
@@ -0,0 +1,25 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (weshaggard) [inline src/Common/test-runtime/XUnit.Runtime.depproj]: I'm putting things like this into under /external directory. I'm fine with it here for now until that comes online though.

File: src/Common/test-runtime/XUnit.Runtime.depproj (hunk 1 above)
Comment (joperezr) [inline src/Common/test-runtime/XUnit.Runtime.depproj]: Agreed. We can add a Condition for test projects so that they don't try to get binplaced.

**Pull request merged**